packages/wonder-blocks-modal/components/modal-backdrop.test.js

```diff
@@ -177,8 +177,57 @@ describe("ModalBackdrop", () => {
             </ModalBackdrop>,
         );
 
-        const focusedElement: HTMLElement = (document.activeElement: any);
-        expect(focusedElement).toBeTruthy();
-        expect(focusedElement.hasAttribute("data-last-button")).toBe(true);
+        const lastButton = wrapper.find("[data-last-button]").getDOMNode();
+        expect(document.activeElement).toBe(lastButton);
+    });
+
+    // TODO(mdr): I haven't figured out how to actually simulate tab keystrokes
+    //     or focus events in a way that JSDOM will recognize, so triggering the
+    //     global focus handler isn't feasible. I had to do manual testing
+    //     instead :( Here's what I had, though!
```

Comment:
According to https://github.com/jsdom/jsdom/issues/1634 some people have been able to trigger keyboard events within JSDOM.  I haven't tried it myself though.

Reply:
This test seems to be sufficient though to test your code.

Reply:
I tried simulating Tab keydown events in similar ways, but it didn't seem to trigger a focus change. It didn't seem to work in Firefox either, so it might be that simulated Tab presses aren't respected? Also might just mean my code was wrong 😬

The test as written doesn't work, even though it seems like it should. First off, I'm not sure that focus handlers would normally trigger synchronously when you call `focus()`. (Maybe?) But, either way, I found that `a.focus()` didn't seem to trigger the global focus event handler, even if I waited a couple seconds. I suspect that the global focus handler doesn't work in JSDOM?

It's also possible that Enzyme doesn't actually mount the component in the global `document`, but instead in its own sandboxed document. So, the listeners we register onto the global document aren't triggered. I hadn't considered that until now!

Reply:
Just spent some time exploring that last theory; no luck. I tried updating `FocusTrap` to register the event on `this._getModalRoot().ownerDocument`, and I wrote a `simulateTabPress` function that triggered the Tab keydown on the button itself, in the correct owner document. I also updated the test to wait a second to see if the focus event happened. Nothing :(

```
function simulateTabPress(target) {
    const event: KeyboardEvent = (target.ownerDocument.createEvent(
        "Event",
    ): any);
    event.key = "Tab";
    event.initEvent("keydown", true, true);
    target.dispatchEvent(event);
}
```